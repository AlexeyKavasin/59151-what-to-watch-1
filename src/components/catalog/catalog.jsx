import React from "react";
import PropTypes from "prop-types";

export const Catalog = ({names, onHeadingClick}) => {
  return (
    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <ul className="catalog__genres-list">
        <li className="catalog__genres-item catalog__genres-item--active">
          <a href="#" className="catalog__genres-link">All genres</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Comedies</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Crime</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Documentary</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Dramas</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Horror</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Kids & Family</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Romance</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Sci-Fi</a>
        </li>
        <li className="catalog__genres-item">
          <a href="#" className="catalog__genres-link">Thrillers</a>
        </li>
      </ul>

      <div className="catalog__movies-list">
        {names.map((name, ind) => {
          return (
            <article className="small-movie-card catalog__movies-card" key={ind}>
              <button className="small-movie-card__play-btn" type="button">Play</button>
              <div className="small-movie-card__image">
                <img src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg" alt={name} width="280" height="175" />
              </div>
              <h3 className="small-movie-card__title" onClick={onHeadingClick}>
                <a className="small-movie-card__link" href="movie-page.html">{name}</a>
              </h3>
            </article>
          );
        })}
      </div>

      <div className="catalog__more">
        <button className="catalog__button" type="button">Show more</button>
      </div>
    </section>
  );
};

Catalog.propTypes = {
  names: PropTypes.array.isRequired,
  onHeadingClick: PropTypes.func
};

Catalog.defaultProps = {
  onHeadingClick: () => {}
};
