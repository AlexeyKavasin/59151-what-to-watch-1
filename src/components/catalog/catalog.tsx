import * as React from "react";
import {connect} from "react-redux";
import GenresList from "../genres-list/genreslist";
import MovieList from "../movie-list/movielist";
import {getCurrentGenre, getAllGenres, filterFilmsByGenre} from "../../redux/reducer/data/selectors.js";
import {withActiveCard} from "../../hocs/with-active-card/with-active-card";
import {selectGenre} from "../../redux/reducer/actions";
import {ICatalog} from "../../interfaces";

const MovieListWithActiveCard = withActiveCard(MovieList);

class Catalog extends React.PureComponent<ICatalog, null> {
  render() {
    const {films, genres, onGenreChange, currentGenre} = this.props;
    return (
      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <GenresList genres={genres} currentGenre={currentGenre} onGenreChange={onGenreChange}/>
        <MovieListWithActiveCard films={films}/>
        <div className="catalog__more">
          <button className="catalog__button" type="button">Show more</button>
        </div>
      </section>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  ownProps,
  currentGenre: getCurrentGenre(state),
  genres: getAllGenres(state),
  films: filterFilmsByGenre(state)
});

const mapDispatchToProps = (dispatch) => ({
  onGenreChange: (genre) => {
    dispatch(selectGenre(genre));
  }
});

export {Catalog};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Catalog);
