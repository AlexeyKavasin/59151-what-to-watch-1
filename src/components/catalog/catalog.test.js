import React from 'react';
import renderer from 'react-test-renderer';
import {Catalog} from './catalog.jsx';

const names = [`Fantastic Beasts`, `Bohemian Rhapsody`, `Macbeth`, `Aviator`];

it(`Catalog renders correctly`, () => {
  const tree = renderer
    .create(<Catalog
      names={names}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
