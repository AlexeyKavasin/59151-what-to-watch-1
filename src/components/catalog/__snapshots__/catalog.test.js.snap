// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Comedies
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Documentary
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Dramas
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Horror
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Kids & Family
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Romance
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Sci-Fi
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Thrillers
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Fantastic Beasts"
          height="175"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Fantastic Beasts
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Bohemian Rhapsody"
          height="175"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Macbeth"
          height="175"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Macbeth
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <button
        className="small-movie-card__play-btn"
        type="button"
      >
        Play
      </button>
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Aviator"
          height="175"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Aviator
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
